src/Symfony/Component/HttpFoundation/JsonResponse.php

```diff
@@ -144,6 +144,10 @@ public function setEncodingOptions($encodingOptions)
     {
         $this->encodingOptions = (int) $encodingOptions;
 
+        if ($this->content !== null) {
+            $this->data = $this->content;
```

Comment:
@stof I think you can check issue 16541 (https://github.com/symfony/symfony/issues/16541) for more details